File: agent/commands-generator/src/main/java/org/matilda/commands/info/ProjectServices.java
package org.matilda.commands.info;

import org.matilda.commands.processors.Processor;

import java.util.List;

public class ProjectServices {
    private final List<ServiceInfo> mServices;

    public ProjectServices(List<ServiceInfo> services) {
        mServices = services;
    }

    public void processEachService(Processor<ServiceInfo> processor) {
        mServices.forEach(processor::process);
    }
}

File: agent/commands-generator/src/main/java/org/matilda/commands/info/ProjectServices.kt
package org.matilda.commands.info

data class ProjectServices(val services: List<StaticServiceInfo>, val dynamicServices: List<ServiceInfo>) {
    fun forEachService(action: (ServiceInfo) -> Unit) {
        forEachStaticService { action(it.serviceInfo) }
        forEachDynamicService(action)
    }

    fun forEachStaticService(action: (StaticServiceInfo) -> Unit) {
        services.forEach(action)
    }

    fun forEachDynamicService(action: (ServiceInfo) -> Unit) {
        dynamicServices.forEach(action)
    }

    fun forEachCommand(action: (CommandInfo) -> Unit) {
        forEachService { it.commands.forEach(action) }
    }

    fun forEachStaticCommand(action: (CommandInfo) -> Unit) {
        forEachStaticService { it.serviceInfo.commands.forEach(action) }
    }
}
